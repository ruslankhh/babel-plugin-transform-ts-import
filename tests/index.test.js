const path = require('path');
const glob = require('glob');
const { transformFileSync } = require('@babel/core');
const plugin = require('../src');

function transform(filePath, options, pluginOptions = {}) {
    return (
        transformFileSync(filePath, Object.assign(
            {
                filename: filePath,
                plugins: [[plugin, pluginOptions]],
                generatorOpts: {
                    comments: false,
                    quotes: 'single',
                    jsescOption: { quotes: 'single' }
                }
            },
            options
        )).code || ''
    );
}

describe('babel-plugin-transform-ts-const', () => {
    glob
        .sync('./fixtures/**/*.js', { cwd: __dirname, dot: false, strict: true })
        .forEach(filePath => {
            it(`transforms ${filePath}`, () => {
                expect(transform(path.join(__dirname, filePath))).toMatchSnapshot();
            });
        });
});
