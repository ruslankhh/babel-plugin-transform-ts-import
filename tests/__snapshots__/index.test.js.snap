// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-ts-const transforms ./fixtures/import-default.js 1`] = `
"'use strict';

var _ref = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    Hello = _ref.default;"
`;

exports[`babel-plugin-transform-ts-const transforms ./fixtures/import-default-and-destruction.js 1`] = `
"'use strict';

var _ref = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    Hello = _ref.default,
    helloPrefix = _ref.helloPrefix;"
`;

exports[`babel-plugin-transform-ts-const transforms ./fixtures/import-destruction.js 1`] = `
"'use strict';

var _ref = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    helloPrefix = _ref.helloPrefix;"
`;

exports[`babel-plugin-transform-ts-const transforms ./fixtures/import-multiple.js 1`] = `
"'use strict';

var _ref = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    Hello = _ref.default;

var _ref2 = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    helloPrefix = _ref2.helloPrefix;

var _ref3 = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    helloPrefix2 = _ref3.helloPrefix2;"
`;

exports[`babel-plugin-transform-ts-const transforms ./fixtures/require-default.js 1`] = `
"'use strict';

var Hello = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}).default;"
`;

exports[`babel-plugin-transform-ts-const transforms ./fixtures/require-default-and-destruction.js 1`] = `
"'use strict';

var _ref = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    helloPrefix = _ref.helloPrefix,
    Hello = _ref.default;"
`;

exports[`babel-plugin-transform-ts-const transforms ./fixtures/require-destruction.js 1`] = `
"'use strict';

var _ref = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    helloPrefix = _ref.helloPrefix;"
`;

exports[`babel-plugin-transform-ts-const transforms ./fixtures/require-multiple.js 1`] = `
"'use strict';

var Hello = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}).default;

var _ref = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    helloPrefix = _ref.helloPrefix;

var _ref2 = function (exports) {
  Object.defineProperty(exports, \\"__esModule\\", {
    value: true
  });
  exports.default = exports.helloPrefix = void 0;
  var helloPrefix = 'Hello';
  exports.helloPrefix = helloPrefix;
  ;

  var Hello = function Hello() {
    var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    if (!props.name) {
      props.name = 'World';
    }

    return ''.concat(helloPrefix, ', ').concat(props.name, '!');
  };

  var _default = Hello;
  exports.default = _default;
  return exports;
}({}),
    helloPrefix2 = _ref2.helloPrefix;"
`;
